### ISSUE TITLE
Invited users for private communities

### ISSUE THREAD

**Issue by flaki** [withspectrum/spectrum#4610]
**Issue Type (check one)**

- [ ] Bug Report
- [x] Feature Idea
- [ ] Technical Discussion
- [ ] Question (these will be auto-closed, please ask them on Spectrum instead https://spectrum.chat/spectrum/open)

**Description (type any text below)**

For larger private communities it could be useful to be able to have users joining "invite only", that is, they would be a member of a specific channel (or channels) they were invited in, but not any other channels, including any non-private channels (unless, again, specifically invited).

In the [TechSpeakers](spectrum.chat/techspeakers) this would be useful for giving a kind of "transient" space for our new members who are currently being onboarded, but not full members until they finish their training. At the end of the training most (but not necessarily all) new members are transitioned into full "member" status so I would imagine these users would start out as a `collaborator` or similar status and could be "upgraded" by admins to a `member` status similar to the UI to how `team` status works today.

This would be super useful for us, but this kind of invite-only collaborator use case I have seen pop up in many other larger organizations I have been seeing.

**Comment by mxstbr** [comment]
Interesting idea!

A quick v1 of this would probably be something like Slack's single-channel guests, would that work for your use case? Have a "newbies" channel that folks-in-training get invited to as single-channel guests, and then transition them to full members when they are ready?

**Comment by flaki** [comment]
Yep, this would cover our use case perfectly, I think that approach would make a great MVP feature for you to figure out if there are any other usecases this would need to be expanded to. :)

**Comment by mxstbr** [comment]
I cannot promise that this is something we will get around to soon, but will keep this in mind!

**Comment by mxstbr** [comment]
I just realised you do not even necessarily need to wait for us to make this happen, you can work around this by creating a private channel (`full-members` or something like that) and then only adding the full members to that private channel!

**Comment by flaki** [comment]
Yeah technically that workaround is possible for communities created from scratch, but for communities that have existed before this need arose that's not an option. Also this kind of "open-by-default" workaround would certainly raise some eyebrows at more privacy/NDA-conscious folks.

This is just to say, I think the initial request is still valid and (considering it should be a relatively small change) still would be something worthwhile to see implemented. But thanks for the tip!
